app/src/androidTest/java/com/github/sdpcoachme/database/CachingStoreTest.kt

```diff
@@ -14,171 +21,189 @@ import com.github.sdpcoachme.data.messaging.Message
-import junit.framework.TestCase.assertEquals
-import junit.framework.TestCase.assertFalse
-import junit.framework.TestCase.assertTrue
+import junit.framework.TestCase.*
+import kotlinx.coroutines.runBlocking
 import org.hamcrest.CoreMatchers.`is`
-import org.hamcrest.MatcherAssert.assertThat
+import org.junit.After
+import org.junit.Before
 import org.junit.Test
 import java.util.concurrent.TimeUnit.SECONDS
 import java.time.DayOfWeek
 import java.time.LocalDate
 import java.time.LocalDateTime
 import java.time.temporal.TemporalAdjusters
 import java.util.concurrent.CompletableFuture
+import java.util.concurrent.TimeUnit.MILLISECONDS
 
-class CachingDatabaseTest {
+val CACHINGSTORE_TEST_PREFERENCES_NAME = "cachingStore_test_preferences"
+val Context.dataStoreTest: DataStore<Preferences> by preferencesDataStore(name = CACHINGSTORE_TEST_PREFERENCES_NAME)
+
+class CachingStoreTest {
 
     // IMPORTANT:
     // Note that here MockDatabase needs to be re-instantiated for each test as we
     // modify its state in the tests.
 
+    private lateinit var wrappedDatabase: Database
+    private lateinit var cachingStore: CachingStore
+
+    @Before
+    fun setUp() {
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        runBlocking {
+            context.dataStoreTest.edit { it.clear() }
+        }
+        wrappedDatabase = MockDatabase()
+        cachingStore = CachingStore(wrappedDatabase,
+            ApplicationProvider.getApplicationContext<Context>().dataStoreTest,
+            ApplicationProvider.getApplicationContext()
+        )
+        cachingStore.retrieveData.get(1, SECONDS)
+    }
+
+    @After
+    fun tearDown() {
+        val context = ApplicationProvider.getApplicationContext<Context>()
+        runBlocking {
+            context.dataStoreTest.edit { it.clear() }
+        }
+    }
+
     @Test
     fun getUserPutsUserInCache() {
-        val wrappedDatabase = MockDatabase()
-        val cachingDatabase = CachingDatabase(wrappedDatabase)
-        cachingDatabase.getUser(exampleEmail).get(5, SECONDS)
-        assertTrue(cachingDatabase.isCached(exampleEmail))
+        cachingStore.getUser(exampleEmail).get(5, SECONDS)
+        assertTrue(cachingStore.isCached(exampleEmail))
     }
 
     @Test
     fun addUserPutsUserInCache() {
-        val wrappedDatabase = MockDatabase()
-        val cachingDatabase = CachingDatabase(wrappedDatabase)
-        val retrievedUser = cachingDatabase.updateUser(willSmithUser)
-            .thenCompose { cachingDatabase.getUser(willSmithUser.email) }
+        val retrievedUser = cachingStore.updateUser(willSmithUser)
+            .thenCompose { cachingStore.getUser(willSmithUser.email) }
             .get(5, SECONDS)
-        assertTrue(cachingDatabase.isCached(willSmithUser.email))
-        assertTrue(cachingDatabase.userExists(willSmithUser.email).get(1, SECONDS))
+        assertTrue(cachingStore.isCached(willSmithUser.email))
```

Comment:
This test does not really test if updateUser caches the user since you always call getUser first and getUser is supposed to cache the user already. So, this would pass even if updateUser did not cache the user since it is already cached at that point. You can have a class that extends the MockDatabase to overwrite updateUser, just like it is done below.